src/requester.ts

```diff
@@ -0,0 +1,89 @@
+import bsv from 'bsv';
+import { AuthHeader, setSignature } from './authentication';
+import { Logger } from './logger';
+import { ErrorNoAdminKey, ErrorResponse } from './errors';
+
+export class Requester {
+  private logger: Logger;
+
+  private adminKey?: bsv.HDPrivateKey;
+  private signingKey?: bsv.HDPrivateKey | bsv.PrivateKey;
+  private xPubString?: string;
+  private url: string;
+
+  static CreateSigningRequester(
+    logger: Logger,
+    url: string,
+    signingKey?: bsv.HDPrivateKey | bsv.PrivateKey,
+    adminKey?: bsv.HDPrivateKey,
+  ) {
+    const requester = new Requester(logger, url);
+    requester.signingKey = signingKey;
+    requester.adminKey = adminKey;
+    return requester;
+  }
+
+  static CreateXPubRequester(logger: Logger, url: string, xPubString: string) {
+    const requester = new Requester(logger, url);
+    requester.xPubString = xPubString;
+    return requester;
+  }
+
+  private constructor(logger: Logger, url: string) {
+    this.logger = logger;
+    this.url = url.endsWith('/') ? url : url + '/'; //make sure the url ends with a '/'
+  }
+
+  async AdminRequest(path: string, method: string = 'GET', payload: any = null): Promise<any> {
```

Comment:
### minor

I'm thinking if it wouldn't be more readable to have something like:
```
async adminGet(...)
async adminPost(...)
```
or even 🙈 to have at the caller side:
```
this.http.admin.get(...);
this.http.get(...);
```